Main.java

```diff
@@ -0,0 +1,88 @@
+public class Main {
+
+    private static final boolean left = true;
+    private static final boolean right = false;
+
+    public static void main(String[] args) {
+
+        // Test the tree from the assignment
+        BinaryTree<Integer> myTree = new BinaryTree(16);
+        myTree.insert(9, 16 , left);
+        myTree.insert(18, 16, right);
+        myTree.insert(3, 9, left);
+        myTree.insert(14, 9, right);
+        myTree.insert(1, 3, left);
+        myTree.insert(5, 3, right);
+        myTree.insert(19, 18, right);
+        printAncestors(myTree, 5);  // 16, 9, 3
+        System.out.println();
+        System.out.println("Lowest common ancestor: " +

```

Comment:
Finding LCA has quite a few edge cases (key not present, one key being ancestor of the other etc), while this println tests only a single one. Additionally it requires a person to look at the output and verify if result is correct. This makes it easy for bugs to slip through.

Could you please add some tests that would check different scenarios, verify results returned by tested methods and output status of the test (passed/failed)? You could either use [JUnit](https://github.com/junit-team/junit4/wiki/Getting-started) or go with your own simple methods.